src/array/cpu/spmm.cc

```diff
@@ -77,6 +77,28 @@ void SpMMCsrHetero(const std::string& op, const std::string& reduce,
   } else if (reduce == "max" || reduce == "min") {
     SWITCH_BITS(bits, DType, {
       SWITCH_OP(op, Op, {
+        std::vector<bool> updated(vec_out.size(), false);
+        // TODO(Israt): use vector updated to fill(out...) too
+        for (dgl_type_t etype = 0; etype < ufeat_node_tids.size(); ++etype) {
+          DType *out_off = vec_out[out_node_tids[etype]].Ptr<DType>();
+          if (reduce == "max")
+            std::fill(out_off, out_off + vec_csr[etype].num_rows * dim, cpu::op::Max<DType>::zero);
+          else
+            std::fill(out_off, out_off + vec_csr[etype].num_rows * dim, cpu::op::Min<DType>::zero);
+          const dgl_type_t dst_id = out_node_tids[etype];
+          if (!updated[dst_id]) {
+            updated[dst_id] = true;
+            // TODO(Israt): what to fill with?
```

Comment:
Remove this comment?  Is there any problem with -1?

Reply:
Don't think so. Etype or node type cannot be -1.